Massive pile of UI changes

mdurrant-b3 opened the pull request:
LBAASv2 UI that actually works.
Needs `api-fix` branch from - https://github.com/mdurrant-b3/neutron-lbaas-dashboard.git for functional testing.
... debating moving this API into a10-horizon until it gets merged in.
Opening it for review to you guys before I take it to Doug and get slaughtered and I'm debating cherry-picking it into smaller PRs.

Commit:
Beginnin gof panel pieces

Commit:
Stuff, I don't care.

Commit:
More stuff.  None of this matters.

Commit:
Even more stuff.

Commit:
More panels. Still needs loading hooks

Commit:
Checkpoint before I switch branches

Commit:
Pruning

Commit:
Beginning of shuffle to make a10_horizon reflect Horizon dir structure
* This has practical implications.  Keeping a similar directory structure allows easier template reuse.
* Instances panel now loads.

Commit:
More shuffling
* a10scaling -> project/slb
* a10ssl -> project/ssl

Commit:
Correct loading of Instances panel

Commit:
Undoing what seemed like a good ideA

Commit:
Restructure Checkpoint
* A10 panel shows up in project and admin, we can now add whatever panels we want to either
* Basic directory structure to support it.

Commit:
Checkpoint commit
* Minimal admin dir structure w/ panels.

Commit:
Admin checkpoint
* All panels display

Commit:
Added overview panel to Admin

Commit:
Checkpoint
* Admin view separation
* Beginning of admin view foundation

Commit:
Updated 'enabled' scripts

Commit:
View/URL consistency

Commit:
checkpoint - beginning of Tables and tabs

Commit:
Panel repair

Commit:
Admin panels that select correctly
* Mock tables/tabs to be populated with real data.

Commit:
Consistent titles

Commit:
Renamed API module for consistency
Added Terminate button (doesn't work yet) to admin instance table

Commit:
Kinda-working delete - it needs an actual backend implementation

Commit:
Overview indexview loads without weird compression error.
Checkpoint commit to change branches to work on user scaffolding.

Commit:
Checkpoint - stuff displaying kinda how I want it

Commit:
Template consolidation

Commit:
Checkpoint - premerge

Commit:
Shared layout templats between admin/user projects.  Yeah!
* Hasn't been applied to all views, did this to make sure it would work.  It works, commit.

Commit:
Additional hooks and update to existing

Commit:
Fixed frustrating cache issue

Commit:
Admin analytics panel

Commit:
Cleanup that happened during troubleshooting.
You can't have panels without views.

Commit:
Checkpoint - URL resolution corrections/table actions

Commit:
Stability commit

Commit:
BP removal

Commit:
Instance creation that needs refactoring

Commit:
Some semblance of instance creation

Commit:
Delete works but doesnt remove the instance becuase instance id doesnt make it up

Commit:
Functional Delete

Commit:
VIP listing/bare updating

Commit:
neutron-lbaas-dashboard requirement for API

Commit:
Checkpoint

isaaczurn committed:
Migrate Button ish

Commit:
API stub

Commit:
Fixes identiy issues along w/ db update

Commit:
Merge checkpoint

Commit:
Merge checkpoint

Commit:
checkpoint

Commit:
pulled in local branches and stash

Commit:
Working merge - jus tsay no to stash

Commit:
Pre pep

Commit:
More PEP

Commit:
Cascading delete from UI

Commit:
Removed PDBs

Commit:
Fixed titles and breadcrumb nav

Commit:
Basic LBaasV2 UI
* supports lbaasv2
* currently requires https://github.com/mdurrant-b3/neutron-lbaas-dashboard/

hthompson6 committed:
Fixed logging

Commit:
PEPoint

Commit:
Another PEP checkpoint

Commit:
PEP checkpoint. Pushing so I can PR

Commit:
PEP and copyright

Cedev commented:
What UI changes are there in this massive pile to test?

How can I test what it would do after merging if there are merge conflicts with master?

Commit:
Fixed monitor type transposition

Commit:
Very simple UI for cert picking/creation - less-than-minimal plumbing

Commit:
Tests for context processors

Commit:
Pulled in branch of fixes and solved stupid workflow issue.

Commit:
Before I break anything

Commit:
Fixed signature issue w/ cert bindings

Commit:
Dont lose stuff

Commit:
Bug fixes
* Correctly restrict deleting of pools and monitors
* Removed "APP_COOKIE" sp_type

Commit:
Fixed LB delete bug.

Commit:
Checkpoint

Commit:
Hid unused project panel scaffolding

Commit:
Added 'Enable SSL' boolean to listener

Commit:
Consistent update titles

Commit:
Consistent checkbox display for multi-actions

Commit:
Beginning of UI abstraction into helper methods
Test fixes

Commit:
Fixed pool display bug
Moved UI helpers
tests for ui helpers
Moved debug/test data to separate file to keep things tidy

Commit:
Member links that don't work

Commit:
Code cleanup
Add member now works from both Pool Detail and Pool Table.
Fixed template paths

Commit:
API Helpers tests running, not passing

Commit:
Fixed Scaling Update action modal titles

Commit:
Fixed revenge of member delete bug

Commit:
Added rollback to listener create for graceful failure
Commented out a-n-l reference to see what breaks

Commit:
optional Description for Certs

Commit:
Cert base table

Commit:
Base table templates
